x/supplier/keeper/unbond_suppliers.go

```diff
@@ -0,0 +1,71 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pokt-network/poktroll/x/shared"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	"github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+// EndBlockerUnbondSupplier unbonds suppliers that have finished the unbonding period.
+func (k Keeper) EndBlockerUnbondSupplier(ctx context.Context) error {
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+
+	logger := k.Logger().With("method", "UnbondSupplier")
+
+	// Iterate over all suppliers and unbond suppliers that have finished the unbonding period.
+	// TODO_IMPROVE: Use an index to iterate over suppliers that have initiated the
+	// unbonding action instead of iterating over all suppliers.
+	for _, supplier := range k.GetAllSuppliers(ctx) {
+		// Ignore suppliers that have not initiated the unbonding action.
+		if supplier.UnstakeCommitSessionEndHeight == types.SupplierNotUnstaking {
+			continue
+		}
+
+		unbondingHeight := k.GetSupplierUnbondingHeight(ctx, &supplier)
+
+		if unbondingHeight <= currentHeight {
```

Comment:
1. Shouldn't this be >= ?
2. IF I'm correct, do we not have integration tests for this?

Reply:
> Shouldn't this be >= ?

We unbond when the unbonding height is anterior (less) than the current height.

> IF I'm correct, do we not have integration tests for this?

We do [x/supplier/keeper/msg_server_unstake_supplier_test.go#L54](https://github.com/pokt-network/poktroll/pull/666/files#diff-bd629e76fe1cba24cc427beb9d08642a2b767cb67f1573e2c2a502cf8e9dd618R54)

Reply:
> Should we use `!supplier.IsActive()` here?

No, because `IsActive` returns `true` for the `Supplier`'s `[stake, unstake]` interval. While here, we check against the `[unstake, proof_window_close]` interval.